lib/events/athena/integration_test.go

```diff
@@ -0,0 +1,391 @@
+	infraOutputs *infraOutputs
+}
+
+func TestIntegrationAthenaSearchSessionEvensBySessionID(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+	defer cancel()
+	ac := setupAthenaContext(t, ctx)
+	eventsSuite := test.EventsSuite{
+		Log:   ac.log,
+		Clock: ac.clock,
+		// Additional 5s is used to compensate for uploading parquet on s3.
+		QueryDelay: ac.batcherInterval + 5*time.Second,
+	}
+
+	eventsSuite.SearchSessionEvensBySessionID(t)
+}
+
+func TestIntegrationAthenaSessionEventsCRUD(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+	defer cancel()
+	ac := setupAthenaContext(t, ctx)
+	eventsSuite := test.EventsSuite{
+		Log:   ac.log,
+		Clock: ac.clock,
+		// Additional 5s is used to compensate for uploading parquet on s3.
+		QueryDelay: ac.batcherInterval + 5*time.Second,
+	}
+
+	eventsSuite.SessionEventsCRUD(t)
+}
+
+func TestIntegrationAthenaEventPagination(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+	defer cancel()
+	ac := setupAthenaContext(t, ctx)
+	eventsSuite := test.EventsSuite{
+		Log:   ac.log,
+		Clock: ac.clock,
+		// Additional 5s is used to compensate for uploading parquet on s3.
+		QueryDelay: ac.batcherInterval + 5*time.Second,
+	}
+
+	eventsSuite.EventPagination(t)
+}
+
+func TestIntegrationAthenaLargeEvents(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+	defer cancel()
+
+	ac := setupAthenaContext(t, ctx)
+	in := &apievents.SessionStart{
+		Metadata: apievents.Metadata{
+			Index: 2,
+			Type:  events.SessionStartEvent,
+			ID:    uuid.NewString(),
+			Code:  strings.Repeat("d", 200000),
+			Time:  ac.clock.Now().UTC(),
+		},
+	}
+	err := ac.log.EmitAuditEvent(ctx, in)
+	require.NoError(t, err)
+
+	// no point of asking before batcher interval.
+	time.Sleep(ac.batcherInterval)
+
+	var history []apievents.AuditEvent
+	// We have batch time 10s, and 5s for upload and additional buffor for s3 download
+	err = retryutils.RetryStaticFor(time.Second*20, time.Second*2, func() error {
+		history, _, err = ac.log.SearchEvents(ac.clock.Now().UTC().Add(-1*time.Minute), ac.clock.Now().UTC(), "", nil, 10, types.EventOrderDescending, "")
+		if err != nil {
+			return err
+		}
+		if len(history) == 0 {
+			return errors.New("events not propagated yet")
+		}
+		return err
+	})
+	require.NoError(t, err)
+	require.Len(t, history, 1)
+	require.Empty(t, cmp.Diff(in, history[0]))
+}
+
+func setupAthenaContext(t *testing.T, ctx context.Context) *athenaContext {
+	testEnabled := os.Getenv(teleport.AWSRunTests)
+	if ok, _ := strconv.ParseBool(testEnabled); !ok {
+		t.Skip("Skipping AWS-dependent test suite.")
+	}
+
+	awsCfg, err := awsconfig.LoadDefaultConfig(ctx)
+	require.NoError(t, err)
+
+	testID := fmt.Sprintf("auditlogs-integrationtests-%v", uuid.New().String())
+
+	clock := clockwork.NewRealClock()
+
+	backend, err := memory.New(memory.Config{})
+	require.NoError(t, err)
+
```

Comment:
yep, I am working on 1 PR that will use locking mechanism on backend so that only 1 auth instace will run batcher, but it will be in separate PR. I have added backend to config too fast I guess.